### ISSUE TITLE
Release prebuild binary of cpp

### ISSUE THREAD

**Issue by stereomatchingkiss** [apache/incubator-mxnet#17003]
## Description
Any plan to release prebuild binary of cpp in the future for different os?Like windows, mac, linux?
It is very difficult to build a workable mxnet.

**Comment by xidulu** [comment]
I believe what you are looking for is the `nightly build`.
You could refer to the `Prerequisites` section in https://github.com/haojin2/PyData-LA-Demo

**Comment by stereomatchingkiss** [comment]
> 
> 
> Do you mean the `nightly build`?
> If that is what you are looking for, you could refer to the `Prerequisites` section in https://github.com/haojin2/PyData-LA-Demo

No, I mean the headers files and the libraries(dll, lib, .a, .so)  which could be used by c++ applications, it is very difficult to build a working c++ library for mxnet on different platforms, every new release of the mxnet may break the old solution which work.

**Comment by leezu** [comment]
I don't think shipping a precompiled libraries is a good solution for C++. It relies on implementation (compiler) defined ABI and may break at any point. You can take a look at [P1863R0](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1863r0.pdf).

Instead, how about we improve the `CMakeLists.txt`, so that MXNet can easily be integrated in other projects that use cmake. https://cliutils.gitlab.io/modern-cmake/ is a good reference. Would that solve your concerns?

**Comment by stereomatchingkiss** [comment]
> 
> 
> I don't think shipping a precompiled libraries is a good solution for C++. It relies on implementation (compiler) defined ABI and may break at any point. You can take a look at [P1863R0](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1863r0.pdf) or listen to https://cppcast.com/titus-winters-abi/.
> 
> Instead, how about we improve the `CMakeLists.txt`, so that MXNet can easily be integrated in other projects that use cmake. https://cliutils.gitlab.io/modern-cmake/ is a good reference. Would that solve your concerns? The idea is that in the end you only need to do add the following line to the `CMakeLists.txt` of your own project: `add_subdirectory(3rdparty/mxnet)` ([reference](https://cliutils.gitlab.io/modern-cmake/chapters/projects/submodule.html))

I hope cmake can make the process easy, but it is still very, very hard to build a cpp library of mxnet after so many years, and I don't think this issue could be solved in the near future.

After a lot of struggle, I manage to build mxnet on ubuntu and windows(especially on windows), the good news are the library works on windows, but do not works on ubuntu.

You could check the issue #16983 if you want to know more details.

**Comment by rioadhii** [comment]
> > Do you mean the `nightly build`?
> > If that is what you are looking for, you could refer to the `Prerequisites` section in https://github.com/haojin2/PyData-LA-Demo
> 
> No, I mean the headers files and the libraries(dll, lib, .a, .so) which could be used by c++ applications, it is very difficult to build a working c++ library for mxnet on different platforms, every new release of the mxnet may break the old solution which work.

Sir, can you share the steps to build MXnet with Blas + CPP_Package on Windows 10 and Visual Studio 2017, it seems the tutorial on the official page doesn't work for me.

Thank you in advance

**Comment by leezu** [comment]
Recommend you use VS 2019

**Comment by rioadhii** [comment]
> Recommend you use VS 2019

I am having trouble with the different cmakelist.txt file, and sadly, it is not according to the tutorial

**Comment by leezu** [comment]
The tutorial may need to be updated. You can also refer to the script used on the CI: https://github.com/apache/incubator-mxnet/blob/master/ci/build_windows.py#L153
